lib/sla.js

```diff
@@ -1,6 +1,93 @@
 'use strict';
+var Q = require('q');
 
-module.exports = function(callback){
-    console.log("\"snarflegluff\" should be printed below:");
-    var str = "snarflegluff";
-    return str;}
+/**
+ * Looks for an existing mixing bridge and creates one if none exist.
+ * @param {Object} client - Object that contains information from the ARI
+ *   connection.
+ * @return {Q} Q promise object.
+ */
+function designateMixingBridge(client) {
+  var defer = Q.defer();
+  var list = Q.denodeify(client.bridges.list.bind(client.bridges));
+  list().then(
+      function(bridges) {
+        var bridge = bridges.filter(function(candidate) {
+          return candidate['bridge_type'] === 'mixing';
+        })[0];
+        if (bridge) {
+          console.log('Using existing mixing bridge %s', bridge.id);
+          defer.resolve(bridge);
+        } else {
+          client.bridges.create({type: 'mixing'}, function(err, bridge) {
```

Comment:
This is still using the callback version of bridges.create and should be switched to promise style.